controllers/ocimachine_controller.go

```diff
@@ -289,6 +289,17 @@ func (r *OCIMachineReconciler) reconcileNormal(ctx context.Context, logger logr.
 			}
 			machineScope.Info("Instance is added to the control plane LB")
 		}
+
+		if len(machine.Spec.VnicAttachments) > 0 {
+			err := machineScope.ReconcileVnicAttachments(ctx)
+			if err != nil {
+				r.Recorder.Event(machine, corev1.EventTypeWarning, "ReconcileError", errors.Wrapf(err, "failed to reconcile OCIMachine").Error())
+				conditions.MarkFalse(machineScope.OCIMachine, infrastructurev1beta1.InstanceReadyCondition, infrastructurev1beta1.InstanceVnicAttachmentFailedReason, clusterv1.ConditionSeverityError, "")
+				return ctrl.Result{}, err
+			}
+			machineScope.Info("Instance vnic attachment success")
```

Comment:
should we add ad codition true here with vnic attachement succeeded?

Reply:
great idea